bitnami/thanos/templates/storegateway/statefulset.yaml

```diff
@@ -237,8 +237,11 @@ spec:
   volumeClaimTemplates:
     - metadata:
         name: data
-        {{- if .Values.storegateway.persistence.annotations }}
-        annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.storegateway.persistence.annotations "context" $) | nindent 10 }}
+        {{- if $.Values.storegateway.persistence.labels }}
+        labels: {{- include "common.tplvalues.render" ( dict "value" $.Values.storegateway.persistence.labels "context" $) | nindent 10 }}
+        {{- end }}
+        {{- if $.Values.storegateway.persistence.annotations }}
+        annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.storegateway.persistence.annotations "context" $) | nindent 10 }}
```

Comment:
Why the leading `$` in the variables?

Reply:
$ is the root context of the values file. It makes sure the references are not affected by context changing in the Helm template.

Reply:
@alemorcuq: Could you please check if this PR needs any improvement?

Reply:
Sorry for the very late response. Yes, the `$` makes sense in the sharded statefulset because we are changing contexts there, but unless I'm missing something the `statefulset.yaml` file doesn't change contexts.

Reply:
Hi @alemorcuq, thank you for your clear response. Indeed you're right, the root context `$` in the `statefulset.yaml` is not necessary. I have updated and tested locally again to confirm it.

Hope that this PR is ready to approval.